test/integration/apiserver/certreload/certreload_test.go

```diff
@@ -253,3 +339,89 @@ YkNtGc1RUDHwecCTFpJtPb7Yu/E=
 		}
 	}
 }
+
+func TestSNICert(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	var servingCertPath string
+
+	_, kubeconfig := framework.StartTestServer(t, stopCh, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			opts.GenericServerRunOptions.MaxRequestBodyBytes = 1024 * 1024
+			servingCertPath = opts.SecureServing.ServerCert.CertDirectory
+
+			if err := ioutil.WriteFile(path.Join(servingCertPath, "foo.key"), anotherServerKey, 0644); err != nil {
+				t.Fatal(err)
+			}
+			if err := ioutil.WriteFile(path.Join(servingCertPath, "foo.crt"), anotherServerCert, 0644); err != nil {
+				t.Fatal(err)
+			}
+
+			dynamiccertificates.FileRefreshDuration = 1 * time.Second
```

Comment:
Seems most tests that care about this value just set the time to 1s like this; clearly the global value could cause problems if tests needed different timings and tests may be in parallel. Not an issue in most cases and each test seems to be setting it as needed instead of assuming defaults. OK/consistent with current practice but wish it wasn't just a global value (not your problem to fix here)